src/server/routes/contract/extensions/accountFactory/write/createAccount.ts

```diff
@@ -63,37 +69,53 @@ export const createAccount = async (fastify: FastifyInstance) => {
       const { simulateTx } = request.query;
       const { adminAddress, extraData, txOverrides } = request.body;
       const {
-        "x-backend-wallet-address": walletAddress,
+        "x-backend-wallet-address": fromAddress,
         "x-account-address": accountAddress,
+        "x-account-factory-address": accountFactoryAddress,
+        "x-account-salt": accountSalt,
         "x-idempotency-key": idempotencyKey,
       } = request.headers as Static<typeof walletWithAAHeaderSchema>;
       const chainId = await getChainIdFromChain(chain);
 
-      const contract = await getContract({
+      const factoryContract = await getContractV5({
         chainId,
-        walletAddress,
         contractAddress,
-        accountAddress,
       });
-      const tx = await contract.accountFactory.createAccount.prepare(
+
+      const deployedAddress = await predictAddress({
+        factoryContract,
         adminAddress,
-        extraData,
-      );
-      const deployedAddress =
-        (await contract.accountFactory.predictAccountAddress(
-          adminAddress,
-          extraData,
-        )) as Address;
+        accountSalt: extraData,
+      });
 
-      const queueId = await queueTx({
-        tx,
-        chainId,
-        simulateTx,
-        extension: "account-factory",
-        deployedContractAddress: deployedAddress,
-        deployedContractType: "account",
-        idempotencyKey,
+      // if extraData is not a hex string, convert it to a hex string
+      // this is the same transformation that is done in the SDK
+      // for predictAddress and createAndSignUserOp
+      // but needed here because we're calling the raw autogenerated abi function
+      const accountSaltHex =
```

Comment:
I assume you want this "hexify if not hex" logic above the `predictAddress` code above.

Reply:
I see, but it would still work to put this higher? It just isn't obvious to someone who doesn't know the SDK behavior that this replicates logic already done in the handler above. 

Alternatively, can createAccount do the same hexify transformation so this logic can be removed?